internal/charmstore/store.go

```diff
@@ -526,9 +526,16 @@ func (s *Store) AddCharm(c charm.Charm, p AddParams) (err error) {
 	if p.URL.URL.Series == "bundle" || p.URL.URL.User == "" || p.URL.URL.Revision == -1 {
 		return errgo.Newf("charm added with invalid id %v", &p.URL.URL)
 	}
-	if p.URL.URL.Series == "" && len(c.Meta().Series) == 0 {
-		return errgo.Newf("charm added without series %v", &p.URL.URL)
+	if len(c.Meta().Series) == 0 {
+		if p.URL.URL.Series == "" {
+			return errgo.WithCausef(nil, params.ErrBadRequest, "charm added without series %v", &p.URL.URL)
```

Comment:
I'm not entirely sure that a Store method should be returning ErrBadRequest. This method doesn't necessarily know that it's being called directly as a result of a request.